moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -236,6 +237,9 @@
   }
 
   private int doPeek() throws IOException {
+    if (stackSize == 1 && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
```

Comment:
What this really wants is `rangeEquals()`: https://github.com/square/okio/issues/165